test/conformance/ptp/ptp.go

```diff
@@ -1081,3 +1138,237 @@ func printInterface(config ptpv1.PtpConfig, interfaceDetailsMap map[string]strin
 		}
 	}
 }
+
+func rebootSlaveNode(fullConfig testconfig.TestConfig) {
+	logrus.Info("Rebooting system starts ..............")
+
+	const (
+		daemonSetName    = "ptp-reboot"
+		containerName    = "container-00"
+		imageWithVersion = "quay.io/testnetworkfunction/debug-partner:latest"
+
+		timeout = 5 * time.Minute
+	)
+
+	// 1. create a daemon set for the node reboot
+	dsRunningPods, err := daemonsets.CreateDaemonSet(daemonSetName, defaultNamespace, containerName, imageWithVersion, timeout)
+	if err != nil {
+		logrus.Errorf("Error : +%v\n", err.Error())
+	}
+	nodeToPodMapping := make(map[string]v1core.Pod)
+	for _, dsPod := range dsRunningPods.Items {
+		nodeToPodMapping[dsPod.Spec.NodeName] = dsPod
+	}
+
+	// 2. Get the slave config, restart the slave node
+	slavePtpConfig := fullConfig.DiscoveredSlavePtpConfig.Config
+
+	restartedNodes := make([]string, len(dsRunningPods.Items))
+
+	for _, recommend := range slavePtpConfig.Spec.Recommend {
+		matches := recommend.Match
+
+		for _, match := range matches {
+			nodeLabel := *match.NodeLabel
+			// Get all nodes with this node label
+			nodes, err := client.Client.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{LabelSelector: nodeLabel})
+			if err != nil {
+				panic(err.Error())
```

Comment:
if you panic here , will it lose all test report?
why can't you fail reboot test in that case?